sdk/communication/azure-communication-phonenumbers/src/test/java/com/azure/communication/phonenumbers/PhoneNumbersAsyncClientIntegrationTest.java

```diff
@@ -0,0 +1,207 @@
+                return result.getFinalResult();
+            }) 
+        ).assertNext((AcquiredPhoneNumber acquiredPhoneNumber) -> {
+            assertEquals(PhoneNumberCapabilityValue.INBOUND_OUTBOUND, acquiredPhoneNumber.getCapabilities().getSms());
+            assertEquals(PhoneNumberCapabilityValue.INBOUND, acquiredPhoneNumber.getCapabilities().getCalling());
+        })
+        .verifyComplete();
+    }
+
+    private PollerFlux<PhoneNumberOperation, PhoneNumberSearchResult> beginSearchAvailablePhoneNumbersHelper(HttpClient httpClient, String testName) {
+        PhoneNumberSearchRequest phoneNumberSearchRequest = new PhoneNumberSearchRequest();
+        PhoneNumberCapabilities capabilities = new PhoneNumberCapabilities();
+        capabilities.setCalling(PhoneNumberCapabilityValue.INBOUND);
+        capabilities.setSms(PhoneNumberCapabilityValue.INBOUND_OUTBOUND);
+        phoneNumberSearchRequest
+            .setAreaCode(AREA_CODE)
+            .setAssignmentType(PhoneNumberAssignmentType.APPLICATION)
+            .setPhoneNumberType(PhoneNumberType.TOLL_FREE)
+            .setCapabilities(capabilities)
+            .setQuantity(1);
+
+        return this.getClientWithConnectionString(httpClient, testName)
+            .beginSearchAvailablePhoneNumbers(COUNTRY_CODE, phoneNumberSearchRequest).setPollInterval(Duration.ofSeconds(1));
+    }
+
+    private PollerFlux<PhoneNumberOperation, Void> beginPurchasePhoneNumbersHelper(HttpClient httpClient, String searchId, String testName) {
+        return this.getClientWithConnectionString(httpClient, testName)
+            .beginPurchasePhoneNumbers(searchId).setPollInterval(Duration.ofSeconds(1));
+    }
+
+    private PollerFlux<PhoneNumberOperation, Void> beginReleasePhoneNumberHelper(HttpClient httpClient, String phoneNumber, String testName) {
+        return this.getClientWithConnectionString(httpClient, testName)
+            .beginReleasePhoneNumber(phoneNumber).setPollInterval(Duration.ofSeconds(1));
+    }
+
+    private PollerFlux<PhoneNumberOperation, AcquiredPhoneNumber> beginUpdatePhoneNumberCapabilitiesHelper(HttpClient httpClient, String phoneNumber, String testName) {
+        PhoneNumberCapabilitiesRequest capabilitiesUpdateRequest = new PhoneNumberCapabilitiesRequest();
+        capabilitiesUpdateRequest.setCalling(PhoneNumberCapabilityValue.INBOUND);
+        capabilitiesUpdateRequest.setSms(PhoneNumberCapabilityValue.INBOUND_OUTBOUND);
+        return this.getClientWithConnectionString(httpClient, testName)
+            .beginUpdatePhoneNumberCapabilities(phoneNumber, capabilitiesUpdateRequest).setPollInterval(Duration.ofSeconds(1));
+    }
+
+    private PhoneNumbersAsyncClient getClientWithConnectionString(HttpClient httpClient, String testName) {
+        PhoneNumbersClientBuilder builder = super.getClientBuilderWithConnectionString(httpClient);
+        return addLoggingPolicy(builder, testName).buildAsyncClient();
+    }
+}
```

Comment:
You may be missing the test for authenticating using managed identity

Reply:
Added, thank you!